### DISCUSSION TITLE
Apostrophe 3.21.0: Scheduled publishing, faster asset builds

### DISCUSSION THREAD

**Discussion by agilbert** [apostrophecms/apostrophe#3774] [Release Notes]
Hi everyone,

This week we are pleased to share a few nice improvements to Apostrophe, including a brand new extension that you can use right away in your projects:

* Our new [Scheduled Publishing](https://github.com/apostrophecms/scheduled-publishing) module is now available to the community, giving your project the ability to schedule documents to be automatically published (or unpublished) in the future.
* You can now unpublish documents manually, making it so they only have a draft state and aren't visible to the public.
* We've added significant improvements to optimize asset build speed in [Apostrophe Assembly](https://apostrophecms.com/assembly) projects using our **multisite** module.
* The UI for the core of ApostropheCMS is now available in French.

Now, as we near the mid-point of the year, we're about to begin work of one of our biggest A3 improvements so far - the addition of a new **advanced permissions** mode where you can create your own user groups with custom permissions. Look out for updates on that soon, and in the meantime, read on for the full release notes below 🙇 

### Adds

* Trigger only the relevant build when in a watch mode (development). The build paths should not contain comma (`,`).
* Adds an `unpublish` method, available for any doc-type.  
An _Unpublish_ option has also been added to the context menu of the modal when editing a piece or a page.
* Adds French translation of the admin UI (use the `fr` locale).

### Fixes

* Vue files not being parsed when running eslint through command line, fixes all lint errors in vue files.
* Fix a bug where some Apostrophe modules symlinked in `node_modules` are not being watched.
* Recover after webpack build error in watch mode (development only).
* Fixes an edge case when failing (throw) task invoked via `task.invoke` will result in `apos.isTask()` to always return true due to `apos.argv` not reverted properly.

## Apostrophe 3.x modules

**[scheduled-publishing](https://github.com/apostrophecms/scheduled-publishing)** 1.0.0

* Command line task to update all documents that have a publishing or unpublishing date by comparing with the current one.
* Since the Home page cannot be unpublished, it's not possible to schedule unpublishing for this specific page.

**multisite** 3.5.0

* Add `APOS_THEME` for single theme build at development startup.
* Add build watching for all available themes in development. Rebuilding happens in the same process and triggers only the relevant themes and builds. Disabling of this feature is possible via `APOS_ASSET_WATCH=0` environment. This feature respects the `APOS_THEME` setting.

## Apostrophe 2.x modules

**[apostrophe-events](https://github.com/apostrophecms/apostrophe-events)** 2.1.8

* Fixes an error that was occurring when trying to save a recurring event.

**Comment by zenflow** [comment]
https://github.com/apostrophecms/scheduled-publishing does not seem to be a public repo.. I get 404

**Reply by boutell** [reply]
This was fixed yesterday. Apologies for the hiccup.

**Reply by zenflow** [reply]
No worries, I wasn't reaching out to use it. Was just reviewing the changelog and noticed that and thought I'd let you know. Excited for my next project using Apostrophe! Keep up the amazing work guys!!

**Reply by boutell** [reply]
Thanks, and we appreciate you letting us know!
